int32[] bounding
int32 stage
int32[] color
int32 open_flag
